gen/string.go

```diff
@@ -70,12 +74,14 @@ func pascalCase(words ...string) string {
 	return strings.Join(words, "")
 }
 
+func constantName(s string) string {
+	return pascalCase(false /* all caps */, strings.Split(s, "_")...)
+}
+
 // enumItemName returns the Go name that should be used for an enum item with
 // the given Thrift name.
 func enumItemName(enumName string, itemName string) string {
-	words := []string{enumName}
-	words = append(words, strings.Split(itemName, "_")...)
-	return pascalCase(words...)
+	return enumName + pascalCase(false /* all caps */, strings.Split(itemName, "_")...)
```

Comment:
enumName is generated by `typeName` which imports the relevant package (if necessary) and returns `foo.Bar` (or just `Bar` if the type is local). I'll add a test for lower case enums.